File: app/src/main/java/org/stypox/dicio/skills/calculator/CalculatorInfo.java
package org.stypox.dicio.skills.calculator;

import static org.stypox.dicio.Sections.isSectionAvailable;
import static org.stypox.dicio.SectionsGenerated.calculator;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import org.stypox.dicio.R;
import org.stypox.dicio.Sections;
import org.dicio.skill.Skill;
import org.dicio.skill.SkillContext;
import org.dicio.skill.SkillInfo;
import org.dicio.skill.chain.ChainSkill;
import org.dicio.skill.standard.StandardRecognizer;

public class CalculatorInfo extends SkillInfo {

    public CalculatorInfo() {
        super("calculator", R.string.skill_name_calculator,
                R.string.skill_sentence_example_calculator, R.drawable.ic_calculate_white, false);
    }

    @Override
    public boolean isAvailable(final SkillContext context) {
        return isSectionAvailable(calculator) && context.getNumberParserFormatter() != null;
    }

    @Override
    public Skill build(final SkillContext context) {
        return new ChainSkill.Builder()
                .recognize(new StandardRecognizer(Sections.getSection(calculator)))
                .process(new CalculatorProcessor())
                .output(new CalculatorOutput());
    }

    @Nullable
    @Override
    public Fragment getPreferenceFragment() {
        return null;
    }
}

File: app/src/main/kotlin/org/stypox/dicio/skills/calculator/CalculatorInfo.kt
package org.stypox.dicio.skills.calculator

import android.content.Context
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Calculate
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.vector.rememberVectorPainter
import org.dicio.skill.context.SkillContext
import org.dicio.skill.skill.Skill
import org.dicio.skill.skill.SkillInfo
import org.stypox.dicio.R
import org.stypox.dicio.sentences.Sentences

object CalculatorInfo : SkillInfo("calculator") {
    override fun name(context: Context) =
        context.getString(R.string.skill_name_calculator)

    override fun sentenceExample(context: Context) =
        context.getString(R.string.skill_sentence_example_calculator)

    @Composable
    override fun icon() =
        rememberVectorPainter(Icons.Default.Calculate)

    override fun isAvailable(ctx: SkillContext): Boolean {
        return Sentences.Calculator[ctx.sentencesLanguage] != null &&
                Sentences.CalculatorOperators[ctx.sentencesLanguage] != null &&
                ctx.parserFormatter != null
    }

    override fun build(ctx: SkillContext): Skill<*> {
        return CalculatorSkill(CalculatorInfo, Sentences.Calculator[ctx.sentencesLanguage]!!)
    }
}
